ironic-config/apache2-ironic-api.conf.j2

```diff
@@ -39,14 +39,15 @@ Listen 6385
     <Directory /usr/bin >
         WSGIProcessGroup ironic
         WSGIApplicationGroup %{GLOBAL}
-        AllowOverride All
-        Require all granted
+        AllowOverride None
         
         {% if "HTTP_BASIC_HTPASSWD" in env and env.HTTP_BASIC_HTPASSWD | length %}
            AuthType Basic
            AuthName "Restricted WSGI area"
            AuthUserFile "/etc/ironic/htpasswd"
            Require valid-user
```

Comment:
The problem is this will cover the endpoints used by IPA (heartbeat etc.) so the deployment will most probably fail. There should not be any authentication on those two endpoints afaik. @dtantsur  Could you please confirm ?

Reply:
I can also confirm it once the CI reaches inspection step

Reply:
```
ci-test-vm-20210906055537-cept:~$ curl -k https://172.22.0.2:6385/v1/nodes
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>401 Unauthorized</title>
</head><body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested.  Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
</body></html>

```
So, as expected this doesn't allow unauthenticated users access to Ironic but it also breaks the inspection as you  can see the in console:

```
  InsecureRequestWarning,
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client [-] An error occurred while attempting to discover the available Ironic API versions, falling back to using version 1.31: json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client Traceback (most recent call last):
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client   File "/usr/lib/python3.6/site-packages/ironic_python_agent/ironic_api_client.py", line 106, in _get_ironic_api_version
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client     data = jsonutils.loads(response.content)
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client   File "/usr/lib/python3.6/site-packages/oslo_serialization/jsonutils.py", line 249, in loads
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client     return json.loads(encodeutils.safe_decode(s, encoding), **kwargs)
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client   File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client     return _default_decoder.decode(s)
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client   File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client   File "/usr/lib64/python3.6/json/decoder.py", line 357, in raw_decode
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client     raise JSONDecodeError("Expecting value", s, err.value) from None
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2021-09-06 06:22:27.681 663 ERROR ironic_python_agent.ironic_api_client 
/usr/lib/python3.6/site-packages/urllib3/connectionpool.py:988: InsecureRequestWarning: Unverified HTTPS request is being made to host '172.22.0.2'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
2021-09-06 06:22:27.818 663 WARNING ironic_python_agent.ironic_api_client [-] Failed looking up node with addresses '00:a0:ef:c2:9e:90,00:a0:ef:c2:9e:8e' at https://172.22.0.2:6385. Error 401: <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>401 Unauthorized</title>
</head><body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested.  Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
</body></html>

```

Reply:
This is funny, but regardless of above logs, I am seeing BMHs in Ready state and and even transitioning to Provisioning  
```
@ci-test-vm-20210906055537-cept:~$ kubectl get bmh -A
NAMESPACE   NAME     STATE          CONSUMER                   ONLINE   ERROR
metal3      node-0   ready                                     true     
metal3      node-1   provisioning   test1-controlplane-qlgcx   true     
```

Reply:
I think the current exceptions are correct here.